app/views/layouts/_settings_item.erb

```diff
@@ -1,6 +1,8 @@
 <% item, link = settings_item %>
 <% selected = (controller.to_s == controller_name.to_s) %>
-<% selected = selected && params[:id] == item.id.to_s  if params[:id] %>
+<% compare_id = item.respond_to?(:id) ? item.id.to_s : '' %>
+<% selected = selected && params[:id] == compare_id  if params[:id] %>
+<% selected = selected && request.fullpath == link if params[:id].blank? %>
```

Comment:
I can't quite make sense of this new logic. What's the new situation that you needed to account for within this partial which led to the change in logic? I tried the PR locally and it's working for me, I'm just thinking about how to potentially simplify but can't quite reason it through yet.

Reply:
It is to cover the case of this partial shown for `["New Assistant", new_settings_assistant_path]` Not sure why the  `['Account', edit_settings_person_path]` doesn't have the same problem